plugins/shared/cmd/launcher/command/device.go

```diff
@@ -190,11 +191,16 @@ func (c *Device) setConfig(spec hcldec.Spec, apiVersion string, config []byte, n
 		return err
 	}
 
-	_, diag, diagErrs := hclutils.ParseHclInterface(configVal, spec, nil)
+	val, diag, diagErrs := hclutils.ParseHclInterface(configVal, spec, nil)
 	if diag.HasErrors() {
 		return multierror.Append(errors.New("failed to parse config: "), diagErrs...)
 	}
 
+	cdata, err := msgpack.Marshal(val, val.Type())
+	if err != nil {
+		return err
+	}
+
 	req := &base.Config{
 		PluginConfig: cdata,
```

Comment:
It's a bit odd that `cdata` variable is used without declaration.  I would have assumed lint-go job would report a warning/error here.  Maybe we need to add a `go install ./...` step.

Reply:
It's (now) declared above, as the output of `msgpack.Marshal`. This was picked up by `lint-go` on master but not before because two different PRs touched the code in incompatible ways; we would have had to rebase the 2nd PR on master to have picked it up in the checks in the PR.

Reply:
Sorry I used wrong tense.  Perfect, sad that we didn't get a merge conflict, but glad that it got detected by CI.